docs/user/instrument-android-crashes-example.md

```diff
@@ -0,0 +1,255 @@
+using the `extras` field.  So if we want to pass along the cause, as well as a few lines of description or even potentially a
+few lines of stack trace, events let us do that easily (with [some limitations](https://mozilla.github.io/glean/book/user/metrics/event.html#limits)).
+
+Now that we have decided what metric type we will use, we can create our `metrics.yaml`.  Inside of the `app` folder in the
+Android Studio project we are going to create a new file and call it `metrics.yaml`.  Then we can add the schema definition and
+the metric definition to the file, so that it looks like this:
+
+```YAML
+# Required to indicate this is a `metrics.yaml` file
+$schema: moz://mozilla.org/schemas/glean/metrics/1-0-0
+
+crash:
+  exception:
+    type: event
+    description: |
+      Event to record crashes caused by unhandled exceptions
+    notification_emails:
+      - crashes@example.com
+    bugs:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1582479
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1582479
+    expires:
+      2099-01-01
+    send_in_pings:
+      - crash
+    extra_keys:
+      cause:
+        description: The cause of the crash
+      message:
+        description: The exception message
+```
+
+By way of explanation, this creates a metric called `exception` within a metric category called `crash`.  There is a brief
+description and the required notification, bug, data review, and expiry fields.  Then we have the `send_in_pings` field with a
+value of `- crash`.  This means that we will send the crash data via a custom ping named `crash` (which we haven’t created,
+yet).  Finally we have the `extra_keys` field with two keys defined, `cause` and `message`.  This will allow us to send a
+couple of pieces of additional information along with the event.
+
+### Add Custom Ping
+
+Now we need to define the custom ping by creating a `pings.yaml` file in the same directory as we just created the `metrics.yaml`
+file.  We already know what the name of the ping is, `crash`, so the `pings.yaml` file should look like this:
+
+```YAML
+# Required to indicate this is a `pings.yaml` file
+$schema: moz://mozilla.org/schemas/glean/pings/1-0-0
+
+crash:
+  description: >
+    A ping to transport crash data
+  include_client_id: true
+  notification_emails:
+    - crash@example.com
+  bugs:
+    - https://bugzilla.mozilla.org/show_bug.cgi?id=1582479
+  data_reviews:
+    - https://bugzilla.mozilla.org/show_bug.cgi?id=1582479
+```
+
+Before we can use our newly defined metric or ping, we need to build the application.  This will cause the [glean_parser](https://github.com/mozilla/glean_parser/)
+to work its magic and generate the API files that represent the metrics we defined.
+
+**NOTE:** Just a little advice from my personal experience here, but I would recommend running the clean task on your project
```

Comment:
IMO this sounds a bit weird in our book (it would be fine in a blog post!)
maybe change that to be a bit more formal:

> **Note**: Currently you should run the `clean` task on your project after every change to ... That makes sure changes are picked up and that the glean_parser regenerates the code.

or similar

Reply:
Plus, I still have hopes that we will be able to fix this after we have the plugin and better input/output specification